dcos/http.py

```diff
@@ -1,11 +1,18 @@
+import getpass
+import sys
+from urlparse import urlparse
+
 import requests
 from dcos import util
 from dcos.errors import DCOSException, DCOSHTTPException
+from requests.auth import HTTPBasicAuth
 
 logger = util.get_logger(__name__)
 
 DEFAULT_TIMEOUT = 5
 
+AUTH_HOSTS = {}  # hostname -> AuthBase()
```

Comment:
That's correct. From RFC1945:

> The realm attribute (case-insensitive) is required for all authentication schemes which issue a challenge. The realm value (case-sensitive), in combination with the canonical root URL of the server being accessed, defines the protection space. These realms allow the protected resources on a server to be partitioned into a set of protection spaces, each with its own authentication scheme and/or authorization database. The realm value is a string, generally assigned by the origin server, which may have additional semantics specific to the authentication scheme.

Not sure what they mean by canonical root URL.